.github/workflows/main.yml

```diff
@@ -0,0 +1,79 @@
+name: run-checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Include all major maintenance + active LTS Node.js versions.
+        # https://github.com/nodejs/Release#release-schedule
+        node: [10, 12, 14]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install xvfb
+          # Ensure both a Chromium and a Firefox binary are available.
+          PUPPETEER_PRODUCT=firefox npm install
+          npm install
+          ls .local-chromium .local-firefox
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Run code checks
+        run: |
+          npm run lint
+          npm run generate-docs
+          npm run ensure-correct-devtools-protocol-revision
+
+      - name: Run unit tests
+        env:
+          CHROMIUM: true
+        run: |
+          xvfb-run --auto-servernum npm run unit
```

Comment:
for my understanding: why do we need xvfb-run now?

Reply:
GitHub Actions workflows run in a headless environment. We need xvfb on Travis CI as well: https://github.com/puppeteer/puppeteer/blob/9a8479a52a7d8b51690b0732b2a10816cd1b8aef/.travis.yml#L2

Reply:
Example error log without xvfb: https://github.com/puppeteer/puppeteer/runs/1668553136#step:6:1050